draft-ietf-tls-tls13.md

```diff
@@ -2310,6 +2311,34 @@ be taken into account when designing new extensions:
   the possibility of version rollback should be a significant consideration in
   any major design change.
 
+
+####  Cookie 
+
+%%% Cookie Extension
+       struct {
+           opaque cookie<0..255>;
+       } Cookie;
+
+Cookies serve two primary purposes:
+
+- Allowing the server to force the client to demonstrate reachability
+  at their apparent network address (thus providing a measure of DoS
+  protection). This is primarily useful for non-connection-oriented
+  transports (see {{?RFC6347}} for an example of this).
+
+- Allowing the server to offload state to the client, thus allowing it to send
+  a HelloRetryRequest without storing any state.
+
+The client MUST provide a Cookie extension in every ClientHello it
+sends. If no cookies are available, the extension MUST be empty (i.e.,
+contain an empty-length cookie, allowing the server to send a Cookie in the
+HelloRetryRequest. A server MAY supply a Cookie in the EncryptedExtensions
+block to allow the client to use it for future connections.
```

Comment:
I see what you are doing.  Until I saw that last bit, I was going to recommend that it MUST NOT be present.  Are you really sure that you want to include the Cookie in subsequent connections?  Won't the PSK ID be enough for that?  Do you think that you need a tracker across connections when PSK isn't in use?